source/common/http/http1/codec_impl.cc

```diff
@@ -473,10 +475,22 @@ void ConnectionImpl::dispatch(Buffer::Instance& data) {
   ssize_t total_parsed = 0;
   if (data.length() > 0) {
     for (const Buffer::RawSlice& slice : data.getRawSlices()) {
-      total_parsed += dispatchSlice(static_cast<const char*>(slice.mem_), slice.len_);
+      dispatching_ = true;
```

Comment:
You might consider using `Cleanup` (or similar) here, to avoid any strangeness around not setting disaptching_ back to false in the early return cases. Granted, it might not matter but probably good to be consistent. If this is done you can probably just set dispatching_ above to true with a guard in a single place?

Reply:
Yes, I like this pattern as well.

Reply:
There is similar code in the H2 codec that should be changed to use cleanup also.

Reply:
Should this also be set during maybeDirectDispatch?  I think it wouldn't hurt.

Setting dispatch_ at the top of ConnectionImpl::dispatch + cleanup to unset seems like the way to go.

Reply:
> Should this also be set during maybeDirectDispatch? I think it wouldn't hurt.

As in, redundantly set to dispatching_ to true inside the method?